src/headers.js

```diff
@@ -27,12 +31,9 @@ class Headers {
   append (name, value) {
     const [normalizedHeaderName, normalizedHeaderValue] = normalizeAndValidateHeaderArguments(name, value)
 
-    const existingHeaderValue = Map.prototype.get.call(this[kHeaders], normalizedHeaderName)
-    if (existingHeaderValue) {
-      Map.prototype.set.call(this[kHeaders], normalizedHeaderName, existingHeaderValue.concat([normalizedHeaderValue]))
-    } else {
-      Map.prototype.set.call(this[kHeaders], normalizedHeaderName, [normalizedHeaderValue])
-    }
+    const existingHeaderValue = (Map.prototype.get.call(this[kHeaders], normalizedHeaderName) || []).slice()
+    existingHeaderValue.push(normalizedHeaderValue)
```

Comment:
This approach over 3 runs before/after yielded about 3 percent perf bonus in the comparative benchmark

.concat |  
-- | --
70.762 | 70.751
70.515 | 70.548
70.614 | 70.59
70.63033333 | 70.62966667
  |  
.push |  
74.343 | 74.382
71.04 | 71.015
74.437 | 74.438
73.27333333 | 73.27833333

Reply:
Probably a fluke since the benchmarks don't use this code 😂